app/controllers/tickets_controller.rb

```diff
@@ -159,26 +159,23 @@ def create
       @ticket = Ticket.new(ticket_params)
     end
 
-    if !@ticket.nil? && @ticket.save
-      NotificationMailer.incoming_message(@ticket, params[:message])
+    if !Ticket.recaptcha_keys_present?
+      ticket_save_and_sent_notification
+    elsif verify_recaptcha
+      ticket_save_and_sent_notification
     end
 
     respond_to do |format|
       format.html do
 
-        if !@ticket.nil? && @ticket.valid?
-
-          if current_user.nil?
-            render 'create'
-          else
-            redirect_to ticket_url(@ticket), notice: I18n::translate(:ticket_added)
-          end
-
+        # if recpatcha is enabled, set conditionals
+        if !Ticket.recaptcha_keys_present?
+          ticket_respond_to_html
+        elsif verify_recaptcha
```

Comment:
I didn't realize that you're requiring signed in users to enter the captcha as well. Could you hide the captcha and not verify it for them? Only not signed in users should enter the captcha as an anti-spam measure.